### DISCUSSION TITLE
docs-multiplayer/netcode/advanced-topics/message-system/custom-messages/index

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#553] [Notes]
# docs-multiplayer/netcode/advanced-topics/message-system/custom-messages/index

A brief explanation of Custom Messages use in Netcode for GameObjects (Netcode) covering Named and Unnamed messages.

https://docs-multiplayer.unity3d.com/netcode/current/advanced-topics/message-system/custom-messages/index.html

**Comment by ghost** [comment]
### Named message

When you received a named message, the `FastBufferReader` has a `ulong` value at the beginning that is the hash of the message name.
Before you read the actual content you want, you need to call `reader.ReadValueSafe(out ulong msgHash)` (if you want the hash) or `reader.Seek(sizeof(ulong))`.
**Note :** this is only true if the client is not the Host/Server. In that case, the hash is not written in the `FastBufferReader`.

_Tested on N4GO 1.2.0._
